modules/core/src/v2/coins/utxo/WalletKeys.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Classes for deriving key triples for wallet addresses.
+ *
+ * By default, BitGo wallets consist of a triple of bip32 extend keypairs.
+ * Every wallet address can be identified by _(chain: number, index: number)_.
+ * The key set for a particular address can be obtained by deriving with the path
+ * `0/0/${chain}/${index}`. (In rare cases the prefix 0/0 can be different)
+ *
+ * Since we never use other derivations for utxo address scripts, the classes defined here only
+ * allow exactly one level of derivation.
+ */
+
+/**
+ * @prettier
+ */
+import * as bip32 from 'bip32';
+import { Triple } from '../../triple';
+
+export function eqPublicKey(a: bip32.BIP32Interface, b: bip32.BIP32Interface): boolean {
+  return a.publicKey.equals(b.publicKey);
+}
+
+/**
+ * Base method for
+ */
+export class WalletKeys {
+  public readonly publicKeys: Triple<Buffer>;
+
+  constructor(public readonly triple: Triple<bip32.BIP32Interface>) {
+    triple.forEach((a, i) => {
+      triple.forEach((b, j) => {
+        if (eqPublicKey(a, b) && i !== j) {
```

Comment:
Just a suggestion but since we always have exactly 3 keys and there are only 3 comparisons necessary I find it a bit more succinct/readable to do
`if(eqPublicKey(this,user, this.backup) || eqPublicKey(this,user, this.bitgo) || eqPublicKey(this,backup this.bitgo))` - but that's just me so feel free to keep as is.

Reply:
3 is right on the threshold ;)

I will leave it as is